### PR TITLE
Jwoo/uwp italic strike through

### PR DESCRIPTION
**Pull request by nesalang**
fixes issue #2963 

issue description:
strike through and italic still existed at UWP API, and old samples cards are needed to be removed.

how verified:
ran UWP test app, and made sure nothing was broken.
ran shared model unit tests.

run textblock and rich textblock json cards on iOS / Android sample apps.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/2970)

### CODE REVIEW

File: samples/v1.2/Elements/TextBlock.Italic.json [hunk 1]
```diff
@@ -1,13 +0,0 @@
-{
-	"type": "AdaptiveCard",
```
Comment (RebeccaAnne) [inline samples/v1.2/Elements/TextBlock.Italic.json]: Please delete the corresponding test files as well. (From source\uwp\AdaptiveCardTestApp\Expected:
del /S *TextBlock.Italic*)

File: source/shared/cpp/ObjectModel/RichTextElementProperties.cpp [hunk 2]
```diff
@@ -30,6 +31,27 @@ Json::Value RichTextElementProperties::SerializeToJsonValue(Json::Value& root) c
...
+{
+    return m_strikethrough;
+}
+
+void RichTextElementProperties::SetStrikethrough(const bool value)
+{
+    m_strikethrough = value;
+}
+
+
```
Comment (RebeccaAnne) [inline source/shared/cpp/ObjectModel/RichTextElementProperties.cpp]: [nit] extra line (consider auto-formatting)

File: source/uwp/AdaptiveCardTestApp/AdaptiveCardTestApp.csproj [hunk 3]
```diff
@@ -149,8 +149,86 @@
     <Content Include="..\..\..\samples\v1.1\Elements\*.json">
       <Link>LinkedCards\v1.1\Elements\%(FileName)</Link>
     </Content>
```
Comment (RebeccaAnne) [inline source/uwp/AdaptiveCardTestApp/AdaptiveCardTestApp.csproj]: Revert this change. It's probably a result of deleting the samples, but we don't want these listed explicitly, we want to include the folder.

File: source/uwp/Renderer/lib/TextHelpers.h [hunk 4]
```diff
@@ -76,26 +80,25 @@ template<typename TXamlTextBlockType>
...
-    boolean italic;
-    RETURN_IF_FAILED(adaptiveTextElement->get_Italic(&italic));
-    if (italic)
+    if (isItalic)
     {
         RETURN_IF_FAILED(xamlTextElement->put_FontStyle(ABI::Windows::UI::Text::FontStyle::FontStyle_Italic));
     }
 
+
     // Get the forground color based on text color, subtle, and container style
```
Comment (RebeccaAnne) [inline source/uwp/Renderer/lib/TextHelpers.h]: [nit] extra line

File: source/uwp/UWPUnitTests/UWPUnitTests.csproj [hunk 5]
```diff
@@ -146,8 +146,86 @@
     <Content Include="..\..\..\samples\v1.1\Elements\*.json">
       <Link>LinkedCards\v1.1\Elements\%(FileName)</Link>
     </Content>
-    <Content Include="..\..\..\samples\v1.2\Elements\*.json">
```
Comment (RebeccaAnne) [inline source/uwp/UWPUnitTests/UWPUnitTests.csproj]: Revert this here as well.

File: source/uwp/AdaptiveCardTestApp/AdaptiveCardTestApp.csproj (hunk 3 above)
Comment (nesalang) [reply source/uwp/AdaptiveCardTestApp/AdaptiveCardTestApp.csproj]: thanks

---
In reply to: [287427936](https://github.com/microsoft/AdaptiveCards/pull/2970#discussion_r287427936) [](ancestors = 287427936)

File: source/uwp/UWPUnitTests/UWPUnitTests.csproj (hunk 5 above)
Comment (nesalang) [reply source/uwp/UWPUnitTests/UWPUnitTests.csproj]: thanks

---
In reply to: [287428525](https://github.com/microsoft/AdaptiveCards/pull/2970#discussion_r287428525) [](ancestors = 287428525)

File: samples/v1.2/Elements/TextBlock.Italic.json (hunk 1 above)
Comment (RebeccaAnne) [reply samples/v1.2/Elements/TextBlock.Italic.json]: There's also a copy of the cards in source\uwp\AdaptiveCardTestApp\Expected\SourceCards

---
In reply to: [287427072](https://github.com/microsoft/AdaptiveCards/pull/2970#discussion_r287427072) [](ancestors = 287427072)
